File: lib/prompts/templates/selfAssessment.ts
import type { Blueprint, GenerateRequest, SectionSpec } from '../../types'

export function buildPrompt(
  spec: SectionSpec,
  blueprint: Blueprint,
  req: GenerateRequest
): string {
  const { lesson_metadata } = req
  return `Generate the self-assessment section for this study guide.

## Blueprint Context
- Lesson Title: ${blueprint.title}
- Learning Targets (use these VERBATIM as skill statements):
${blueprint.learning_targets.map((lt) => `  ${lt.number}. ${lt.objective}`).join('\n')}
- Subject: ${lesson_metadata.subject}
- Grade Level: ${lesson_metadata.grade_level}

## Instructions
Create a self-assessment table where:
1. Each row corresponds to one learning target from the blueprint
2. The "skill" field must be VERBATIM from blueprint.learning_targets[].objective
3. Each row has columns: needs_assistance, minimal_understanding, confident (these are checkbox columns — do not add descriptive text, just the column name)
4. Include exactly ${blueprint.learning_targets.length} rows, one per learning target

Also include the 4 reflection stems EXACTLY as specified below — do not change any word.

## Output Schema
{
  "self_assessment": {
    "rows": [
      {
        "skill": "string (verbatim from blueprint learning_targets[].objective)",
        "columns": ["needs_assistance", "minimal_understanding", "confident"]
      }
    ]
  },
  "reflection_stems": [
    "I find _____ the most interesting because _____.",
    "I need to improve on _____ because _____.",
    "I need to practice _____ because _____.",
    "I plan to _____."
  ]
}

Section spec: ${JSON.stringify(spec)}`
}

File: backend/study-guide-agent/app/prompts/templates/self_assessment.py
from __future__ import annotations

from app.types import Blueprint, GenerateRequest


def build_prompt(spec, blueprint: Blueprint, request: GenerateRequest) -> str:
    del spec, request

    self_assessment_schema = """{
    "title": "Self-Assessment",
    "confidence_levels": [
        "Not yet",
        "Getting there",
        "Confident"
    ],
    "rows": [
        {
            "skill": "string",
            "reflection_prompt": "string"
        }
    ]
}"""

    target_text = "\n".join(
        f"- {target.objective}" for target in blueprint.learning_targets
    )

    prompt_lines = [
        "Create the self-assessment section for a K-12 study guide.",
        f"- Lesson title: {blueprint.title}",
        f"- Essential question: {blueprint.essential_question}",
        "- Learning target objectives that must be used verbatim as skill values:",
        target_text,
        "Requirements:",
        "- Create one row per blueprint learning target, in the same order.",
        "- Each rows.skill value must match the corresponding learning target objective verbatim.",
        "- Use exactly these confidence levels: Not yet, Getting there, Confident.",
        "- Add a brief reflection_prompt that helps a learner judge their own evidence of success.",
        "- Do not rewrite, shorten, or paraphrase the skill text.",
        "Expected JSON schema:",
        self_assessment_schema,
        "Return only JSON.",
    ]

    return "\n".join(prompt_lines)
